src/util/iroha-util.js

```diff
@@ -0,0 +1,555 @@
+        const request = new pbEndpoint.TxStatusRequest()
+
+        request.setTxHash(txHash)
+
+        txClient.status(request, (err, response) => {
+          if (err) {
+            return reject(err)
+          }
+
+          const status = response.getTxStatus()
+          const TxStatus = require('iroha-lib/pb/endpoint_pb.js').TxStatus
+          const statusName = getProtoEnumName(
+            TxStatus,
+            'iroha.protocol.TxStatus',
+            status
+          )
+
+          if (statusName !== 'COMMITTED') {
+            return reject(new Error(`Your transaction wasn't commited: expected=COMMITED, actual=${statusName}`))
+          }
+
+          resolve()
+        })
+      })
+    })
+}
+
+/**
+ * transferAsset https://hyperledger.github.io/iroha-api/#transfer-asset
+ * @param {String} srcAccountId
+ * @param {String} destAccountId
+ * @param {String} assetId
+ * @param {String} description
+ * @param {String} amount
+ */
+function transferAsset (srcAccountId, destAccountId, assetId, description, amount) {
+  debug('starting transferAsset...')
+
+  let txClient, txHash
+
+  return new Promise((resolve, reject) => {
+    const tx = txBuilder
+      .creatorAccountId(cache.username)
+      .txCounter(1)
+      .createdTime(Date.now())
+      .transferAsset(srcAccountId, destAccountId, assetId, description, amount)
+      .build()
+    const protoTx = makeProtoTxWithKeys(tx, cache.keys)
+
+    txClient = new endpointGrpc.CommandServiceClient(
+      cache.nodeIp,
+      grpc.credentials.createInsecure()
+    )
+    txHash = blob2array(tx.hash().blob())
+
+    debug('submitting transaction...')
+    debug('peer ip:', cache.nodeIp)
+    debug('parameters:', JSON.stringify(protoTx.toObject().payload, null, '  '))
+    debug('txhash:', Buffer.from(txHash).toString('hex'))
+    debug('')
+
+    txClient.torii(protoTx, (err, data) => {
+      if (err) {
+        return reject(err)
+      }
+
+      debug('submitted transaction successfully!')
+      resolve()
+    })
+  })
+    .then(() => {
+      debug('sleep 5 seconds...')
+      return sleep(5000)
+    })
+    .then(() => {
+      debug('sending transaction status request...')
+
+      return new Promise((resolve, reject) => {
+        const request = new pbEndpoint.TxStatusRequest()
+
+        request.setTxHash(txHash)
+
+        txClient.status(request, (err, response) => {
+          if (err) {
+            return reject(err)
+          }
+
+          const status = response.getTxStatus()
+          const TxStatus = require('iroha-lib/pb/endpoint_pb.js').TxStatus
+          const statusName = getProtoEnumName(
+            TxStatus,
+            'iroha.protocol.TxStatus',
+            status
+          )
+
+          if (statusName !== 'COMMITTED') {
+            return reject(new Error(`Your transaction wasn't commited: expected=COMMITED, actual=${statusName}`))
+          }
+
+          resolve()
+        })
+      })
+    })
+}
+
+/*
+ *  ===== utilities ===
```

Comment:
Maybe we should transfer utilities to the separate file, so we don't have that large "iroha utils"? 🤔

Reply:
Later on maybe it makes sense to move those utils into "iroha-lib"? @stinger112 what do you think?

Reply:
> Maybe we should transfer utilities to the separate file, so we don't have that large "iroha utils"? 🤔
I'm not sure whether we should separate these utilities to another file because these functions are used only in iroha-util. However, it's also true that iroha-util.js is quite large. 😅